salt/auth/ldap.py

```diff
@@ -352,11 +357,11 @@ def groups(username, **kwargs):
             search_results = bind.search_s(search_base,
                                            ldap.SCOPE_SUBTREE,
                                            search_string,
-                                           [_config('accountattributename'), 'cn'])
+                                           [salt.utils.stringutils.to_str(_config('accountattributename')), str('cn')])
 
             for entry, result in search_results:
                 for user in result[_config('accountattributename')]:
-                    if username == user.split(',')[0].split('=')[-1]:
+                    if username == salt.utils.data.decode(user).split(',')[0].split('=')[-1]:
```

Comment:
If I'm not wrong `salt.utils.stringutils.to_unicode` shall be used here instead of `salt.utils.data.decode`.
Probably @terminalmage will correct me.